pandas/tests/indexes/test_multi.py

```diff
@@ -766,6 +766,14 @@ def test_from_arrays_empty(self):
                 ValueError, "Must pass non-zero number of levels/labels"):
             MultiIndex.from_arrays(arrays=[])
 
+        # 0 levels, names defined
+        result = MultiIndex.from_arrays(arrays=[],
```

Comment:
The above is due to conversions that happen in `_factorize_from_iterables` that do not occur in `__new__`. I think it makes sense for an empty multi-index to not have base type, and the code I originally wrote in `__new__` ignores the indexes created in `from_arrays`. If someone wanted to create an empty multi-index with typed levels, however, this could break that.